src/violations.rs

```diff
@@ -0,0 +1,5689 @@
+use itertools::Itertools;
+
+use crate::define_violation;
+use crate::flake8_debugger::types::DebuggerUsingType;
+use crate::flake8_pytest_style::types::{
+    ParametrizeNameType, ParametrizeValuesRowType, ParametrizeValuesType,
+};
+use crate::flake8_quotes::settings::Quote;
+use crate::flake8_tidy_imports::settings::Strictness;
+use crate::pyupgrade::types::Primitive;
+use crate::registry::{
+    Branch, DeferralKeyword, EqCmpop, IsCmpop, LiteralType, MockReference, UnusedCodes,
+};
+use crate::violation::{AlwaysAutofixableViolation, Violation};
+
+// pycodestyle errors
+
+define_violation!(
+    pub struct MultipleImportsOnOneLine;
+);
+impl Violation for MultipleImportsOnOneLine {
+    fn message(&self) -> String {
+        "Multiple imports on one line".to_string()
+    }
+
+    fn placeholder() -> Self {
+        MultipleImportsOnOneLine
```

Comment:
Can these references be `Self` instead? It looks like it from my testing.

Reply:
Yes they can be. However I think it makes more sense to get rid of `placeholder` entirely instead. We only use it for documentation and I think it often does a very poor job at that e.g. `'...' % ... expected mapping but got sequence` or `... is banned: ...` really are not informative and looking at the placeholder message often is confusing since it's not clear what is part of the message and what is interpolated.

I think as soon as we have a system where the rules can be documented via Rust doc comments on the structs we can look into getting rid of placeholder in favor of the new doc comments.

So I think you can add "Document rules via doc comments on violation structs" to your list of things to do for future PRs :)

If we still want to document the message format strings I think that's best achieved via a proc macro. In any case I don't think hand-written placeholder implementations make much sense.

Reply:
I agree. Though, of course, we do need the placeholders _for now_ until we have an alternative.